### PR TITLE
Remove legacy L1T digis dependencies in Online DQM

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

In the ticket [CMSALCAFAST-94](https://its.cern.ch/jira/browse/CMSALCAFAST-94), while testing a new Global Tag candidate meant for a general cleaning of unused / previous runs unmaintained records, few online DQM clients (`dt4ml`, `dt`, `ecal`, and `hlt`) encountered issues and failed.
The reason of these failures was analyzed as unnecessary dependencies of the current DQM clients on the legacy L1T digis.
This is fixed / worked around in this PR.
   * 9b3afd7e478fd1ec4b0b66f876e1188754feab27 removes the dependency in `HLTSiStripMonitorCluster` by means of re-configuring the few instances of `GenericTriggerEventFlag` class to use the stage 2 gt digis in presence of the `stage2L1Trigger` modifier.
   * 8c7fd9b32506c8450f6734d9201da13714317043 replaces the legacy gt digis with the stage2 ones in presence of the `stage2L1Trigger` modifier
   * fe46246e44fbac0a69747078acf9ac64a72d68e1 works around the fact that the `DQM/EcalMonitorTasks/src/ClusterTask.cc` natively doesn't support the stage2 L1T setup but requires products from the legacy L1T (this is followed up also at https://github.com/cms-sw/cmssw/issues/46715) 

#### PR validation:

The following commands work with this branch,

```
scram b runtests_TestDQMOnlineClient-hlt_dqm_sourceclient use-ibeos
scram b runtests_TestDQMOnlineClient-dt_dqm_sourceclient use-ibeos
scram b runtests_TestDQMOnlineClient-dt4ml_dqm_sourceclient use-ibeos
scram b runtests_TestDQMOnlineClient-ecal_dqm_sourceclient use-ibeos
```

when tested by using the Global Tag proposed at  [CMSALCAFAST-94](https://its.cern.ch/jira/browse/CMSALCAFAST-94).

```diff
diff --git a/DQM/Integration/python/config/FrontierCondition_GT_cfi.py b/DQM/Integration/python/config/FrontierCondition_GT_cfi.py
index 5845afa34bb..a91ce205422 100644
--- a/DQM/Integration/python/config/FrontierCondition_GT_cfi.py
+++ b/DQM/Integration/python/config/FrontierCondition_GT_cfi.py
@@ -1,7 +1,7 @@
 import FWCore.ParameterSet.Config as cms
 from Configuration.StandardSequences.FrontierConditions_GlobalTag_cff import *
 from Configuration.AlCa.autoCond import autoCond
-GlobalTag.globaltag = autoCond['run3_hlt']
+GlobalTag.globaltag = "141X_dataRun3_HLT_Candidate_2024_11_15_CleaningOfRecords"  #autoCond['run3_hlt']
 
 #############################################
 #
```

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by antoniovagnerini** [comment]
+1
- spurious differences from HGCal DQM test (open issue  #46416)

**Comment by mandrenguyen** [comment]
+1

File: DQM/Integration/python/clients/dt4ml_dqm_sourceclient-live_cfg.py [hunk 1]
```diff
@@ -117,6 +118,7 @@
     process.twinMuxStage2Digis.DTTM7_FED_Source = "rawDataRepacker"
     process.dtunpacker.inputLabel = "rawDataRepacker"
     process.gtDigis.DaqGtInputTag = "rawDataRepacker"
+    process.gtStage2Digis.InputLabel = "rawDataCollector"
```
Comment (perrotta) [inline DQM/Integration/python/clients/dt4ml_dqm_sourceclient-live_cfg.py:121]: Shouldn't it be here:
```PYTHON
    process.gtStage2Digis.InputLabel = "rawDataRepacker"
```

Comment (mmusich) [reply DQM/Integration/python/clients/dt4ml_dqm_sourceclient-live_cfg.py:121]: I think it can be whatever. It is not in any path apparently.

**Pull request merged**